### ISSUE TITLE
[chunkhash] substitution in spriteFilename config option

### ISSUE THREAD

**Issue by kisenka** [kisenka/svg-sprite-loader#98]

**Comment by obartra** [comment]
Are you looking for contributions regarding this issue?

**Comment by kisenka** [comment]
@obartra contributions is always very welcome :)

**Comment by obartra** [comment]
I've submitted https://github.com/kisenka/svg-sprite-loader/pull/100

**Comment by kisenka** [comment]
@obartra please check `svg-sprite-loader@beta`, I've implemented `[spritehash]` placeholder, check [corresponding test](https://github.com/kisenka/svg-sprite-loader/blob/e712692aaf97b777d43c544d6e71772f96b80dd4/test/loader.test.js#L306).

**Comment by kisenka** [comment]
BTW it is also possible to use sprite hash as query param and with combination with `[chunkname]`, e.g.:
```js
module.exports = {
  module: {
    rules: [
      {
        test: /\.svg$/,
        loader: 'svg-sprite-loader',
        options: {
          spriteFilename: '[chunkname].svg?[spritehash:6]'
        }
      }
    ]
  }
}
```

**Comment by kisenka** [comment]
@obartra it works? closing this one?

**Comment by obartra** [comment]
Hey there, awesome job with the beta! it is working well.

The hash varies per build, am I correct in assuming it's not content-based?

**Comment by kisenka** [comment]
> The hash varies per build, am I correct in assuming it's not content-based?

Are you sure? It should be hash sum of the sprite symbols content - [here](https://github.com/kisenka/svg-sprite-loader/blob/8f08971d6ed457a3cb9ff54fcf30b800f239a157/lib/utils/mapped-list.js#L113). And [here](https://github.com/kisenka/svg-sprite-loader/blob/8f08971d6ed457a3cb9ff54fcf30b800f239a157/lib/utils/hash-tokens.js#L21) is an interpolate function. You can test it separetly:

```js
const hashTokens = require('svg-sprite-loader/lib/utils/hash-tokens');

const res = hashTokens('[chunkname]-[spritehash:6]', { spritehash: '<svg>...</svg>' });
console.log(res);
```

**Comment by kisenka** [comment]
BTW do we also need `[chunkhash]` and `[fillhash]` token?

**Comment by obartra** [comment]
Let me double check, since I'm generating the svgs dynamically there could be something else going on but I was getting different generated hashes when using `spritehash` on successive builds.

**Comment by kisenka** [comment]
@obartra ping

**Comment by obartra** [comment]
I could be doing something wrong but even when I disable all loaders (except for `svg-sprite-loader`) and plugins (except for `SpriteLoaderPlugin`) I still get different results on different runs.

My setup is fairly small as well, this is what I'm doing:

```javascript
        include: join(basePath, '/static/images/svg'),
        loader: 'svg-sprite-loader',
        options: {
          extract: true,
          spriteFilename: 'icon-sprite-[spritehash:6].svg'
        }
```

**Comment by kisenka** [comment]
@obartra very strange. I need repo with reproducible behaviour :)

**Comment by obartra** [comment]
I'm using it in a private repo, let me see if I can get a demo ready

**Comment by kisenka** [comment]
Shipped with `svg-sprite-loader@2.1.0`, `[hash]` should be used instead of `[spritehash]`
